doc/whats_new/v1.0.rst

```diff
@@ -59,6 +59,13 @@ Changelog
 - |Fix| Fixes incorrect multiple data-conversion warnings when clustering
   boolean data. :pr:`19046` by :user:`Surya Prakash <jdsurya>`.
 
+:mod:`sklearn.compose`
+......................
+
+- |Enhancement| :meth:`compose.ColumnTransformer.transform` now only requires
+  non-dropped columns to appear, regardless of the order. :pr:`19263` by
+  `Thomas Fan`_
```

Comment:
I find this to be very hard to read without context. Let's make it "allows" rather than "requires". Is the following an accurate description of the change?

```suggestion
- |Enhancement| :class:`compose.ColumnTransformer` now allows DataFrame input to 
  have its columns appear in a changed order in `transform`. Further, columns that
  are dropped will not be required in transform, and additional columns will be
  ignored if `remainder='drop'`. :pr:`19263` by `Thomas Fan`_
```